htlcswitch/interfaces.go

```diff
@@ -180,3 +180,31 @@ type TowerClient interface {
 	// isTweakless should be true.
 	BackupState(*lnwire.ChannelID, *lnwallet.BreachRetribution, bool) error
 }
+
+// htlcNotifier is an interface which represents the input side of the
+// HTLCNotifier which htlc events are piped through. This interface is intended
+// to allow for mocking of the htlcNotifier in tests, so is unexported because
+// it is not needed outside of the htlcSwitch package.
+type htlcNotifier interface {
```

Comment:
Because the interface is actually only required to make tests in link/switch easier. The implementation is set in the server so needs to be exported.